bes_publisher/buildevent/service.go

```diff
@@ -146,9 +199,58 @@ func (b *buildStream) handleEvent(req *bpb.PublishBuildToolEventStreamRequest) e
 		return err
 	}
 
-	// TODO(scott): insert this message into PubSub
 	glog.V(2).Infof("# Bazel event:\n%s", prototext.Format(event))
-	metricBuildEventServiceEventCount.WithLabelValues(oneofType(event.Payload), "ok").Inc()
+	b.updateAttrs(event)
+	if err := b.maybePublish(event); err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateAttrs updates the attribute set that is sent out with each pubsub
+// message.
+func (b *buildStream) updateAttrs(event *bes.BuildEvent) {
+	switch payload := event.Payload.(type) {
+	case *bes.BuildEvent_Started:
+		b.attrs["inv_id"] = payload.Started.GetUuid()
+	}
+}
+
+// maybePublish publishes the given event if it is one that we care about;
+// otherwise, the event is dropped.
+func (b *buildStream) maybePublish(event *bes.BuildEvent) error {
```

Comment:
nit: the `maybe` prefix feels odd. down below you have a func called `recordErrFrom` yet it doesn't always record an error. Maybe a more descriptive name here is `handleBuildEvent` or something more descriptive to the action the method is doing.

Reply:
I figured `recordErrFrom` is named as such because it always records errors (though non-errors are ignored); `maybePublish` may or may not publish an event, even if it is present